### ISSUE TITLE
Analyzing Block Part Propagation Time and Its Impact on Network Throughput

### ISSUE THREAD

**Issue by staheri14** [celestiaorg/celestia-core#1257]
As part of big block analysis, it is advantageous to ascertain the duration required for each proposal (block parts) to be disseminated across the network. This knowledge is crucial for gauging the fraction of block time allocated to the propagation of block parts and understanding how enhancements in block propagation speed could affect block time and, consequently, network throughput. 

This task is intended to monitor the advancement of this analysis.

**Comment by staheri14** [comment]
I have extracted the results, but the plots appear somewhat inconsistent and do not match my theoretical expectations. Specifically, the block heights for the block propagation speed, which are measured using round state tracing data, are off by approximately one height. Going to investigate it.


In the figure presented, the block gossip time represents the average duration from the receipt of proposal to the submission of prevote for all peers (99 peers in control-1 experiment) at each block height.


![output](https://github.com/celestiaorg/celestia-core/assets/35961250/549af541-b00d-4b58-9276-ddb3cbe98f98)

**Comment by evan-forbes** [comment]
> are off by approximately one height

yeah the block time is from the time in the commit (last commit) so its off by one

**Comment by evan-forbes** [comment]
what does the script do for heights with multiple rounds?

**Comment by staheri14** [comment]
> > are off by approximately one height
> 
> yeah the block time is from the time in the commit (last commit) so its off by one

I did not see any fix in the main branch, that's why I made this [PR](#1259)

**Comment by staheri14** [comment]
> what does the script do for heights with multiple rounds?

In this particular plot, it only captures the first round of a multi-round height. I have an extended version for height, round combinations.

**Comment by staheri14** [comment]
When I shift the height manually, the results will be like below, however, if could have updated data it would be more accurate:

![time-size-gossiptime](https://github.com/celestiaorg/celestia-core/assets/35961250/8a531b5e-90e9-4743-8050-e74bc9610d4e)

**Comment by staheri14** [comment]
Sharing more results:

- The ratio of the block propagation time to the block time 
![ratio](https://github.com/celestiaorg/celestia-core/assets/35961250/37b55b9a-4971-4a7e-aa54-7cae626c6f15)

Based on this new plot, block propagation takes almost 60-80 percent of the block time.

**Comment by evan-forbes** [comment]
per the two node followup that was just published, I think we actually have to dig a bit deeper to find the actually block gossip time. This is because the above data collection do not account for the time spent in Prepare or Process Proposal

**Comment by evan-forbes** [comment]
@staheri14 , do you mind clarifying what is meant by this 

> ascertain the duration required for each proposal (block parts) to be disseminated across the network

is this the same thing as the network trace?

**Comment by staheri14** [comment]
> @staheri14 , do you mind clarifying what is meant by this
> 
> > ascertain the duration required for each proposal (block parts) to be disseminated across the network
> 
> is this the same thing as the network trace? or is this blocked by the network trace?

Sure, this is how I calculated it:  the time between the receipt of a block proposal until the node issues a prevote for that proposal (indicating that the entire proposal has been dowloaded).
